pkgs/games/anki/default.nix

```diff
@@ -1,197 +1,397 @@
-    substituteInPlace anki/consts.py \
-      --replace 'HELP_SITE="http://ankisrs.net/docs/manual.html"' \
-                'HELP_SITE="${manual}/share/doc/anki/html/manual.html"'
-  '';
+  inherit yarnOfflineCache;
 
-  # UTF-8 locale needed for testing
-  LC_ALL = "en_US.UTF-8";
+  buildPhase = ''
+    export RUST_BACKTRACE=1
+    export RUST_LOG=debug
+    mkdir -p out
+    echo ${builtins.substring 0 8 rev} > out/buildhash
+    touch out/env
+    mkdir -p out/rust/debug
+    ln -vsf ${pyEnv} ./out/pyenv
+    ln -vsf ${pyEnv} ./out/pyenv-qt5
+    ln -vsf ${anki-runner}/bin/runner ./out/rust/debug/runner
+    ln -vsf ${anki-configurator}/bin/configure ./out/rust/debug/configure
+    ln -vsf ${fakeArchive}/bin/archives ./out/rust/debug/archives
+    mkdir -p out/extracted/node/bin
+    ln -vsf ${offlineYarn}/bin/yarn out/extracted/node/bin/yarn
+    mkdir -p .git
+    touch .git/HEAD
+    mkdir -p out/rslib/i18n
+    ln -vsf ${anki-i18n} out/rslib/i18n/strings.json
+    rsync -avP ${anki-nodemodules}/ out/node_modules/
+    chmod +w ./out/node_modules
 
-  # tests fail with to many open files
-  doCheck = !stdenv.isDarwin;
+    mkdir -p out/extracted/python/libs
+    ln -vsf ${anki-rsbridge}/lib/* out/extracted/python/libs/
+    mkdir -p out/rust/release
+    ln -vsf ${anki-rsbridge}/lib/* out/rust/release/librsbridge.so
 
-  # - Anki writes some files to $HOME during tests
-  # - Skip tests using network
-  checkPhase = ''
-    HOME=$TMP pytest --ignore tests/test_sync.py
-  '';
+    export HOME=$NIX_BUILD_TOP
+    yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+    fixup_yarn_lock yarn.lock
 
-  installPhase = ''
-    pp=$out/lib/${python.libPrefix}/site-packages
-
-    mkdir -p $out/bin
-    mkdir -p $out/share/applications
-    mkdir -p $doc/share/doc/anki
-    mkdir -p $man/share/man/man1
-    mkdir -p $out/share/mime/packages
-    mkdir -p $out/share/pixmaps
-    mkdir -p $pp
-
-    cat > $out/bin/anki <<EOF
-    #!${python}/bin/python
-    import aqt
-    aqt.run()
-    EOF
-    chmod 755 $out/bin/anki
-
-    cp -v anki.desktop $out/share/applications/
-    cp -v README* LICENSE* $doc/share/doc/anki/
-    cp -v anki.1 $man/share/man/man1/
-    cp -v anki.xml $out/share/mime/packages/
-    cp -v anki.{png,xpm} $out/share/pixmaps/
-    cp -rv locale $out/share/
-    cp -rv anki aqt web $pp/
-
-    # copy the manual into $doc
-    cp -r ${manual}/share/doc/anki/html $doc/share/doc/anki
+    mkdir -p out/pylib/anki
+    mkdir -p out/wheels/
+
+    ${anki-configurator}/bin/configure
+    PIP_USER=1 ${anki-runner}/bin/runner build wheels
   '';
 
-  # now wrapPythonPrograms from postFixup will add both python and qt env variables
-  dontWrapQtApps = true;
+  doCheck = false;
+  preInstall = ''
+    set -x
+    mkdir dist
+    mv out/wheels/* dist
+    export PYTHONPATH=$PYTHONPATH:${python39.pkgs.protobuf}/lib/python3.9/site-packages
+  '';
 
-  preFixup = ''
-    makeWrapperArgs+=(
```

Comment:
I'll re-add this so we aren't double-wrapping before going out of draft.

I'll also add in `lame` and `mpv`, I'm pretty sure they're still necessary.

Reply:
done